apps/sms/js/subject_composer.js

```diff
@@ -106,7 +113,8 @@
 
       // methods
       updateValue: updateValue.bind(this),
-      updatePlaceholder: updatePlaceholder.bind(this)
+      updatePlaceholder: updatePlaceholder.bind(this),
+      updateDirection: updateDirection.bind(this)
```

Comment:
maybe if you don't `bind` you can access the `input` directly in `updateDirection` using `this.input` ?